features/fixtures/mazerunner/jvm-scenarios/src/main/java/com/bugsnag/android/mazerunner/scenarios/CrashHandlerScenario.kt

```diff
@@ -22,9 +24,13 @@ internal class CrashHandlerScenario(
         val previousHandler = Thread.getDefaultUncaughtExceptionHandler()
 
         Thread.setDefaultUncaughtExceptionHandler({ t, e ->
-            Log.d("Bugsnag", "Intercepted uncaught exception")
+            Log.d("Bugsnag", "CrashHandlerScenario: Intercepted uncaught exception")
+            loggedMessages.add("CrashHandlerScenario: Intercepted uncaught exception")
             previousHandler.uncaughtException(t, e)
         })
         throw RuntimeException("CrashHandlerScenario")
     }
+
+    override fun getInterceptedLogMessages() =
```

Comment:
While this should work, there are a couple of enhancements I'd propose which are generally useful when writing Kotlin:

1. `val` can be preferred over `var` because it means the property won't change from its initial value, resulting in less moving parts
2. `List` should be preferred over `MutableList` as it guarantees the elements won't change after initialization

With that in mind I'd suggest the following approach:

```suggestion
    override fun getInterceptedLogMessages() = listOf("CrashHandlerScenario: Intercepted uncaught exception")
```

Reply:
I'm not sure I understand this approach, as it feels like it's shortcutting the actual test.
In this case only adding the log to the list once the error handler fires would be the whole point?

Reply:
Ah, this was a misunderstanding on my part, I'll be resolving this shortly.